### PR TITLE
wrapper

### PR DESCRIPTION
**Pull request by vhnatyk**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+**/*.verifier*
+**/*.ledger*
+**/*.operator*
+**/*.crawler*
+**/*.synthetic_node*
+**/*.github*
+**universal.srs.*.*
+!**universal.srs.*.metadata
+*.cargo*
+**/*.log*
```
Comment (ImmanuelSegol) [inline .gitignore:27]: add new line.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "snarkVM"]
+	path = snarkVM
+	url = git@github.com:AleoHQ/snarkVM.git
```
Comment (ImmanuelSegol) [inline .gitmodules:3]: just asking do we need a specific version or just latests?

File: src/main.rs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+async fn do_puzzle<N: Network>(
+    args: Args,
+    coinbase_puzzle: CoinbasePuzzle<N>,
+    challenge: Challenge<N>,
+    address: Address<N>,
+) {
+    let threads_num = args.threads_num;
+    let parallel_num = args.parallel_num;
+
+    let mut thread_pools = Vec::new();
```
Comment (ImmanuelSegol) [inline src/main.rs:167]: with_capacity is way faster if u know the thread_pools size

File: src/main.rs [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+    let mut provers_state = start_provers(thread_pools.clone(), coinbase_puzzle.clone(), challenge.clone(), address);
+
+    loop {
+        std::thread::sleep(Duration::from_millis(5000));
+
+        let task_id = provers_state.task_id.clone();
+        let elapsed = provers_state.proves_start.read().unwrap().elapsed().as_secs_f64();
+        let iterations = provers_state.proves_count.load(Ordering::SeqCst) as usize;
+        if elapsed >= 60f64 {
```
Comment (ImmanuelSegol) [inline src/main.rs:191]: `60f64` can this be constant with an explanation plz

File: .gitmodules (hunk 2 above)
Comment (vhnatyk) [reply .gitmodules:3]: latest is ok

File: src/main.rs (hunk 4 above)
Comment (vhnatyk) [reply src/main.rs:191]: y, will do

File: src/main.rs (hunk 3 above)
Comment (vhnatyk) [reply src/main.rs:167]: if the size of array is really big) I think if size isn't really big, it's some nanos

Comment (ImmanuelSegol) [reply src/main.rs:167]: oh ok ok

**Pull request merged**